services/subnet_manager/src/main/java/com/futurewei/alcor/subnet/config/RedisConfiguration.java

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2019 The Alcor Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+        you may not use this file except in compliance with the License.
+        You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+        Unless required by applicable law or agreed to in writing, software
+        distributed under the License is distributed on an "AS IS" BASIS,
+        WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+        See the License for the specific language governing permissions and
+        limitations under the License.
+*/
+
+package com.futurewei.alcor.subnet.config;
+
+import com.futurewei.alcor.common.service.RedisListener;
+import com.futurewei.alcor.subnet.entity.SubnetState;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.domain.EntityScan;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.data.redis.connection.RedisStandaloneConfiguration;
+import org.springframework.data.redis.connection.lettuce.LettuceConnectionFactory;
+import org.springframework.data.redis.core.RedisTemplate;
+import org.springframework.data.redis.listener.ChannelTopic;
+import org.springframework.data.redis.listener.RedisMessageListenerContainer;
+import org.springframework.data.redis.listener.adapter.MessageListenerAdapter;
+import org.springframework.data.redis.serializer.Jackson2JsonRedisSerializer;
+import org.springframework.data.redis.serializer.StringRedisSerializer;
+
+@Configuration
+@ComponentScan({"com.futurewei.alcor.subnet.service", "com.futurewei.alcor.common.service"})
+@EntityScan({"com.futurewei.alcor.subnet.entity}", "com.futurewei.alcor.common.entity"})
+public class RedisConfiguration {
```

Comment:
Do we need a dedicated RedisConfiguration file every microservice, or could share one file in the /lib folder?

Reply:
I think it is better to keep it in every microservce because they have different redis-host and redis-post

Reply:
Got it thanks